File: Python/백준/Silver/2839. 설탕 배달/설탕 배달.py
n = int(input())

bag_count = 0

while n >= 0:
    if n % 5 == 0:
        bag_count += n // 5
        print(bag_count)
        break

    bag_count += 1
    n = n - 3
else:
    # 정확히 나눌 수 없는 경우
    print(-1)

File: Java/백준/Silver/2839. 설탕 배달/설탕 배달.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        int n = scanner.nextInt();
        scanner.close();

        int bagCount = 0;

        while (n >= 0) {
            if (n % 5 == 0) {
                bagCount += n / 5;
                System.out.println(bagCount);
                return;
            }
            bagCount++;
            n = n - 3;  // 3kg 봉지 하나 추가
        }

        // 정확히 나눌 수 없는 경우
        System.out.println(-1);
    }
}